app/assets/javascripts/govuk_publishing_components/components/cookie-banner.js

```diff
@@ -101,6 +101,7 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
   CookieBanner.prototype.showConfirmationMessage = function () {
     this.$cookieBannerMainContent = document.querySelector('.js-banner-wrapper')
 
+    this.$cookieBannerMainContent.style.display = 'none'
```

Comment:
Rather than doing this in the JavaScript, would adding:

```css
[hidden] {
  display: none;
}
```
to the CSS work?

Reply:
This should all already be handled by this code in GOV.UK Frontend:

https://github.com/alphagov/govuk-frontend/blob/72e9d6e43cffa7db79477343a2cf931812f9b7c7/src/govuk/components/cookie-banner/_index.scss#L20-L24

I think the issue is that you're showing and hiding the cookie banner by setting `display: block` rather than by toggling the `hidden` attribute. The `display: block;` inline styles is then overriding the CSS.

If you consistently use the `hidden` attribute then this should all 'just work'.

There's a 'test' client-side implementation here if that helps (it's a lot simpler as it doesn't actually read or store cookies):

https://github.com/alphagov/govuk-frontend/blob/72e9d6e43cffa7db79477343a2cf931812f9b7c7/app/views/full-page-examples/cookie-banner-client-side/index.njk#L184-L223

Reply:
According to caniuse `hidden` isn't supported by IE10? https://caniuse.com/hidden

Reply:
IE10 and older have no _native_ support for the attribute, but the `[hidden]` attribute selector works just fine in CSS, which means we can make it work with just:

```css
.govuk-cookie-banner[hidden] {
  display: none;
}
```

The implementation in GOV.UK Frontend was tested all the way back to IE8 as part of https://github.com/alphagov/govuk-frontend/pull/2126.

Reply:
thanks all, sounds like this needs some reworking!